apps/src/code-studio/pd/workshop_dashboard/components/session_form_part.jsx

```diff
@@ -39,17 +49,14 @@ var SessionFormPart = React.createClass({
     onChange: React.PropTypes.func.isRequired,
   },
 
-  handleDateChange(e) {
-    this.props.session.date = e.target.value;
-    this.props.onChange(this.props.session);
-  },
-  handleStartTimeChange(e) {
-    this.props.session.startTime = e.target.value;
-    this.props.onChange(this.props.session);
-  },
-  handleEndTimeChange(e) {
-    this.props.session.endTime = e.target.value;
-    this.props.onChange(this.props.session);
+  handleChange(path, value) {
+    let updatedSession = _.set(
```

Comment:
This could be const instead of let.

JS's version of const is much weaker than in most languages. Essentially it means "I can't reuse this variable name in this scope". So the following things are completely valid:
```
const foo = { x: 1, y: 2};
foo.x = 3; // the object isn't frozen, even though it's const
```

```
const x = 1;
function bar(valueThatWasConst) {
  valueThatWasConst = valueThatWasConst + 1;
}
bar(x);